Zend/zend_strtod.c

```diff
@@ -533,37 +545,28 @@ Bigint {
 
  typedef struct Bigint Bigint;
 
+#ifndef Bigint
  static Bigint *freelist[Kmax+1];
+#endif
 
 static void destroy_freelist(void);
 static void free_p5s(void);
 
-#ifdef ZTS
+#ifdef MULTIPLE_THREADS
 static MUTEX_T dtoa_mutex;
 static MUTEX_T pow5mult_mutex;
 #endif /* ZTS */
 
```

Comment:
```suggestion
```

You forgot to remove these?

Reply:
And the now obsolete usages of the acquire/release macros. I suppose you just did the minimum to draft this for now.

Reply:
I didn't plan to remove this and the acquire/release macros as they are part of the "API" of the file, which appears to be a reusable piece of code imported from elsewhere. These macros are documented at the beginning of the file: https://github.com/php/php-src/blob/077891fccd2879ac95630c7c4b8c494819569fd1/Zend/zend_strtod.c#L147-L155

There are many knobs like this in this file, many of which we will never use, like `KR_headers`.

So I only removed the definition of MULTIPLE_THREADS, and left the default no-op definitions of ACQUIRE_DTOA_LOCK and FREE_DTOA_LOCK.

I don't mind removing their use as well if you think it's better.

I feel that we should eventually replace this code by more modern implementations of strtod and dtoa. It should be possible to implement these without memory allocations. Also I don't know if we still need to support VAX/IBM arithmetic. This feels risky and largely out of scope of this PR however.